no version available for transitive dependencies

**Describe the bug**
I use Quarkus with quarkus-web-bundler and when I trying to import:

		<dependency>
			<groupId>org.mvnpm</groupId>
			<artifactId>primereact</artifactId>
			<version>10.8.4</version>
			<scope>provided</scope>
		</dependency>

then I get this error:

```
Could not collect dependencies for project com.acme:hello:jar:1.0.0-SNAPSHOT
[ERROR] No versions available for org.mvnpm:scheduler:jar:[16.6.2,17) within specified range
[ERROR] No versions available for org.mvnpm:scheduler:jar:[16.6.2,17) within specified range
```

**To Reproduce**
Steps to reproduce the behavior:
1. Add the following to pom.xml:

```
<!-- Add quarkus 3.16.1 -->
..
<dependency>
    <groupId>io.quarkiverse.web-bundler</groupId>
    <artifactId>quarkus-web-bundler</artifactId>
    <version>1.7.3</version>
</dependency>
<dependency>
    <groupId>org.mvnpm</groupId>
    <artifactId>primereact</artifactId>
    <version>10.8.4</version>
    <scope>provided</scope>
</dependency>
```

3. Update maven deps
4. See error

**Expected behavior**
The lib should be downloaded without error.

**Additional context**
I used mvnpm for the first time. I don't know if this is really a bug or if I have to take care of downloading the lib `scheduler` myself but I expect the specified libraries in mvnpm (in this case: primereact) to take care of transitive deps themselves.

--- a/pom.xml
+++ b/pom.xml
@@ -53,7 +53,7 @@
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
     <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
     <quarkus.platform.group-id>io.quarkus.platform</quarkus.platform.group-id>
-    <quarkus.platform.version>3.14.1</quarkus.platform.version>
+    <quarkus.platform.version>3.16.2</quarkus.platform.version>
     <skipITs>true</skipITs>
     <surefire-plugin.version>3.2.5</surefire-plugin.version>
     <formatter.plugin.version>2.23.0</formatter.plugin.version>
@@ -62,19 +62,19 @@
     
     <pgpainless.version>1.5.5</pgpainless.version>
     <!-- UI Libs -->
-    <quarkus-web-bundler.version>1.7.2</quarkus-web-bundler.version>
-    <lit.version>3.2.0</lit.version>
-    <vaadin.version>24.4.5</vaadin.version>
+    <quarkus-web-bundler.version>1.7.3</quarkus-web-bundler.version>
+    <lit.version>3.2.1</lit.version>
+    <vaadin.version>24.5.3</vaadin.version>
     <ldrs.version>1.0.2</ldrs.version>
     <vaadin-router.version>1.7.5</vaadin-router.version>
     <codeblock.version>1.0.16</codeblock.version>
-    <compare-versions.version>6.1.0</compare-versions.version>
-    <marked.version>12.0.1</marked.version>
+    <compare-versions.version>6.1.1</compare-versions.version>
+    <marked.version>15.0.0</marked.version>
     <card.version>1.0.1</card.version>
     <badge.version>1.0.2</badge.version>
     <!-- Testing-->
     <playwright.version>0.0.1</playwright.version>
-    <esbuild-java.version>1.5.0</esbuild-java.version>
+    <esbuild-java.version>1.5.1</esbuild-java.version>
   </properties>
   <dependencyManagement>
     <dependencies>

--- a/src/main/java/io/mvnpm/JarContentsApi.java
+++ b/src/main/java/io/mvnpm/JarContentsApi.java
@@ -9,6 +9,8 @@ import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.jar.JarFile;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 import jakarta.inject.Inject;
 import jakarta.ws.rs.GET;
@@ -33,6 +35,8 @@ import io.mvnpm.maven.UrlPathParser;
  */
 @Path("/api")
 public class JarContentsApi {
+
+    private static final Logger LOGGER = Logger.getLogger(JarContentsApi.class.getName());
 
     @Inject
     MavenRepositoryService mavenRepositoryService;
@@ -77,17 +81,15 @@ public class JarContentsApi {
         library.setVersion(nameVersionType.version());
         library.setType(filetype.getPostString());
 
+        Map<String, JarAsset> assetMap = new HashMap<>();
+        JarAsset rootAsset = new JarAsset();
+        rootAsset.setName("/");
+        rootAsset.setFileAsset(false);
+        rootAsset.setChildren(new ArrayList<>());
+        assetMap.put("/", rootAsset);
+
+        // Iterate through the entries of the jar file
         try (JarFile jarFile = new JarFile(path.toString())) {
-            Map<String, JarAsset> assetMap = new HashMap<>();
-
-            // Create a root asset
-            JarAsset rootAsset = new JarAsset();
-            rootAsset.setName("/");
-            rootAsset.setFileAsset(false);
-            rootAsset.setChildren(new ArrayList<>());
-            assetMap.put("/", rootAsset);
-
-            // Iterate through the entries of the jar file
             jarFile.stream().forEach(entry -> {
                 String name = entry.getName();
                 boolean isFile = !entry.isDirectory();
@@ -129,7 +131,7 @@ public class JarContentsApi {
             library.setRootAsset(rootAsset);
 
         } catch (IOException e) {
-            e.printStackTrace();
+            LOGGER.log(Level.SEVERE, "Failed to load JAR file: " + path, e);
         }
 
         return library;
@@ -157,9 +159,9 @@ public class JarContentsApi {
             ArchiveEntry entry;
 
             while ((entry = tarIn.getNextEntry()) != null) {
-                if (!tarIn.canReadEntryData(entry)) {
+                if (!tarIn.canReadEntryData(entry))
                     continue;
-                }
+
                 String name = entry.getName();
                 boolean isFile = !((TarArchiveEntry) entry).isDirectory();
                 String[] parts = name.split("/");
@@ -198,7 +200,7 @@ public class JarContentsApi {
             }
             library.setRootAsset(rootAsset);
         } catch (IOException e) {
-            e.printStackTrace();
+            LOGGER.log(Level.SEVERE, "Failed to load TAR.GZ file: " + path, e);
         }
 
         return library;

--- a/src/main/java/io/mvnpm/composite/CompositeCreator.java
+++ b/src/main/java/io/mvnpm/composite/CompositeCreator.java
@@ -202,63 +202,60 @@ public class CompositeCreator {
             Files.createDirectories(outputJar.getParent());
             // Create a new JAR file to merge the others into
             Name outputJarName = NameParser.fromMavenGA(model.getGroupId(), model.getArtifactId());
-            try (OutputStream jarFile = Files.newOutputStream(outputJar);
-                    JarOutputStream mergedJar = new JarOutputStream(jarFile)) {
 
+            try (OutputStream jarFile = Files.newOutputStream(outputJar);
+                    JarOutputStream mergedJar = new JarOutputStream(jarFile);
+                    ByteArrayOutputStream commonTgzBaos = new ByteArrayOutputStream();
+                    GZIPOutputStream commonTgzGzos = new GZIPOutputStream(commonTgzBaos);
+                    TarArchiveOutputStream commonTgzOut = new TarArchiveOutputStream(commonTgzGzos)) {
+
+                commonTgzOut.setLongFileMode(TarArchiveOutputStream.LONGFILE_POSIX);
                 // Collect all importmaps to merge at the end
                 Map<String, String> importmaps = new HashMap<>();
                 // Create new merged pom.xml
                 pom.getDependencies().clear();
                 Map<String, Dependency> newDependencies = new HashMap<>();
                 Map<String, Developer> newDevelopers = new HashMap<>();
-                Map<String, License> newLicences = new HashMap<>();
+                Map<String, License> newLicenses = new HashMap<>();
+                int countMvnpmMoreTgz = 0;
 
-                try (ByteArrayOutputStream commonTgzBaos = new ByteArrayOutputStream();
-                        GZIPOutputStream commonTgzGzos = new GZIPOutputStream(commonTgzBaos);
-                        TarArchiveOutputStream commonTgzOut = new TarArchiveOutputStream(commonTgzGzos)) {
-                    commonTgzOut.setLongFileMode(TarArchiveOutputStream.LONGFILE_POSIX);
-                    int countMvnpmMoreTgz = 0;
-                    for (Dependency dependency : dependencies) {
-                        Name jarName = NameParser.fromMavenGA(dependency.getGroupId(), dependency.getArtifactId());
-                        Path jarPath = mavenRepositoryService.getPath(jarName, dependency.getVersion(), FileType.jar);
+                for (Dependency dependency : dependencies) {
+                    Name jarName = NameParser.fromMavenGA(dependency.getGroupId(), dependency.getArtifactId());
+                    Path jarPath = mavenRepositoryService.getPath(jarName, dependency.getVersion(), FileType.jar);
 
-                        try (InputStream inputStream = Files.newInputStream(jarPath);
-                                JarInputStream inputJar = new JarInputStream(inputStream)) {
-                            // Add all entries from the input JAR to the merged JAR
-                            JarEntry entry;
-                            while ((entry = inputJar.getNextJarEntry()) != null) {
+                    try (InputStream inputStream = Files.newInputStream(jarPath);
+                            JarInputStream inputJar = new JarInputStream(inputStream)) {
+                        // Add all entries from the input JAR to the merged JAR
+                        JarEntry entry;
+                        while ((entry = inputJar.getNextJarEntry()) != null) {
+                            String entryName = entry.getName();
 
-                                String entryname = entry.getName();
-
-                                if (entryname.equals("META-INF/importmap.json")) {
-                                    // Remember importmap
-                                    importmaps.putAll(getImportMap(inputJar));
-                                } else if (entryname.startsWith("META-INF/maven")) {
-                                    updatePom(inputJar, newDependencies, newDevelopers, newLicences, entryname,
-                                            List.copyOf(mapByGA(dependencies).keySet()));
-                                } else if (entryname.startsWith(MVNPM_MORE_ARCHIVE)) {
-                                    countMvnpmMoreTgz++;
-                                    extractTgzEntriesAndMergeToCommon(inputJar, commonTgzOut);
-                                } else if (!entryname.endsWith("LICENSE")) { // Ignore we will add one in root
-                                    writeEntry(inputJar, mergedJar, entry);
-                                }
+                            if ("META-INF/importmap.json".equals(entryName)) {
+                                // Remember importmap
+                                importmaps.putAll(getImportMap(inputJar));
+                            } else if (entryName.startsWith("META-INF/maven")) {
+                                updatePom(inputJar, newDependencies, newDevelopers, newLicenses, entryName,
+                                        List.copyOf(mapByGA(dependencies).keySet()));
+                            } else if (entryName.startsWith(MVNPM_MORE_ARCHIVE)) {
+                                countMvnpmMoreTgz++;
+                                extractTgzEntriesAndMergeToCommon(inputJar, commonTgzOut);
+                            } else if (!entryName.endsWith("LICENSE")) {
+                                writeEntry(inputJar, mergedJar, entry);
                             }
-                        } catch (IOException | XmlPullParserException e) {
-                            throw new RuntimeException(e);
                         }
-                    }
-                    commonTgzOut.finish();
-                    commonTgzOut.close();
-                    commonTgzGzos.finish();
-                    commonTgzGzos.close();
-                    if (countMvnpmMoreTgz > 0) {
-                        writeEntry(mergedJar, MVNPM_MORE_ARCHIVE, commonTgzBaos.toByteArray());
+                    } catch (IOException | XmlPullParserException e) {
+                        throw new RuntimeException("Error processing JAR entry: " + jarPath, e);
                     }
                 }
 
+                commonTgzOut.finish();
+                if (countMvnpmMoreTgz > 0) {
+                    writeEntry(mergedJar, MVNPM_MORE_ARCHIVE, commonTgzBaos.toByteArray());
+                }
+
                 // Add importmap (in jar and on disk)
-                Aggregator a = new Aggregator(importmaps);
-                String aggregatedImportMap = a.aggregateAsJson(false);
+                Aggregator aggregator = new Aggregator(importmaps);
+                String aggregatedImportMap = aggregator.aggregateAsJson(false);
                 writeEntry(mergedJar, "META-INF/importmap.json", aggregatedImportMap);
                 Path importmapPath = getImportMapPath(outputJarName, model.getVersion());
                 fileStore.createFile(importmapPath, aggregatedImportMap.getBytes());
@@ -269,7 +266,7 @@ public class CompositeCreator {
                 Path outputPom = Path.of(pomName);
                 pom.setDependencies(List.copyOf(newDependencies.values()));
                 pom.setDevelopers(List.copyOf(newDevelopers.values()));
-                pom.setLicenses(List.copyOf(newLicences.values()));
+                pom.setLicenses(List.copyOf(newLicenses.values()));
                 MavenXpp3Writer mxw = new MavenXpp3Writer();
 
                 FileUtil.createDirectories(outputPom);
@@ -281,29 +278,31 @@ public class CompositeCreator {
                 }
 
                 // Add pom.properties
-                Properties p = new Properties();
-                p.setProperty("groupId", pom.getGroupId());
-                p.setProperty("artifactId", pom.getArtifactId());
-                p.setProperty("version", pom.getVersion());
+                Properties properties = new Properties();
+                properties.setProperty("groupId", pom.getGroupId());
+                properties.setProperty("artifactId", pom.getArtifactId());
+                properties.setProperty("version", pom.getVersion());
                 try (StringWriter writer = new StringWriter()) {
-                    p.store(writer, "Generated by mvnpm.org");
-                    String pomProperties = writer.toString();
+                    properties.store(writer, "Generated by mvnpm.org");
                     writeEntry(mergedJar, "META-INF/maven/" + pom.getGroupId() + "/" + pom.getArtifactId() + "/pom.properties",
-                            pomProperties);
+                            writer.toString());
                 }
 
                 Log.info(pom.getGroupId() + ":" + pom.getArtifactId() + ":" + pom.getVersion() + " created");
-            }
-            // Also create the Sha1
-            FileUtil.createSha1(outputJar);
 
-            // Also kick off all other files creation
+            } catch (IOException e) {
+                throw new RuntimeException("Error creating JAR output: " + outputJar, e);
+            }
+
+            // Create SHA-1 and other files
+            FileUtil.createSha1(outputJar);
             fileStore.touch(outputJarName, model.getVersion(), outputJar);
         }
+
         return pom;
     }
 
-    private static void extractTgzEntriesAndMergeToCommon(JarInputStream inputJar, TarArchiveOutputStream commonTgz)
+    private void extractTgzEntriesAndMergeToCommon(JarInputStream inputJar, TarArchiveOutputStream commonTgz)
             throws IOException {
         try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
             IOUtils.copy(inputJar, baos);

--- a/src/main/java/io/mvnpm/mavencentral/SonatypeFacade.java
+++ b/src/main/java/io/mvnpm/mavencentral/SonatypeFacade.java
@@ -102,7 +102,7 @@ public class SonatypeFacade {
 
             if (authorization.isPresent()) {
                 String a = "Basic " + authorization.get();
-                //byte[] b = Files.readAllBytes(path);
+
                 Response uploadResponse = sonatypeClient.uploadBundle(a, path);
 
                 if (uploadResponse.getStatus() == 201) {

--- a/src/main/java/io/mvnpm/version/VersionConverter.java
+++ b/src/main/java/io/mvnpm/version/VersionConverter.java
@@ -48,7 +48,6 @@ public class VersionConverter {
     }
 
     public static String convert(String versionString) {
-        System.out.println("> CONVERT " + versionString);
         try {
             if (null == versionString || versionString.startsWith("git:/") || versionString.startsWith("git+http")) { // We do not support git repos as version. Maybe something we can add later
                 versionString = EMPTY;
@@ -71,9 +70,7 @@ public class VersionConverter {
     private static String convertMultiple(String[] versions) {
         List<String> versionList = new ArrayList<>();
         for (String v : versions) {
-            System.out.println(">> MULTIPLE " + v);
             String result = convertMultiplePart(v.trim()).trim();
-            System.out.println("<< result " + result);
             versionList.add(result);
         }
 
@@ -102,34 +99,29 @@ public class VersionConverter {
 
         // Hyphen range
         if (version.contains(SPACE + HYPHEN + SPACE)) {
-            System.out.println(">>> Hyphen " + version);
             return convertHyphen(version);
         }
 
         // Tilde range
         if (version.startsWith(TILDE)) {
-            System.out.println(">>> Tilde " + version);
             version = cleanTildeVersion(version);
             return convertTilde(version);
         }
 
         // Caret range
         if (version.startsWith(CARET)) {
-            System.out.println(">>> Caret " + version);
             version = cleanCaretVersion(version);
             return convertCaret(version);
         }
 
         // Operator range
         if (version.startsWith(LESS_THAN) || version.startsWith(GREATER_THAN)) {
-            System.out.println(">>> Operator " + version);
             version = cleanOperatorVersion(version);
             return convertOperator(version);
         }
 
         // X range
         if (!version.contains(SPACE)) {
-            System.out.println(">>> Xrange " + version);
             String numberPart = version.split(HYPHEN)[0];
             if (numberPart.contains(STAR) || numberPart.contains(EX) || numberPart.contains(EX.toUpperCase())) {
                 return convertX(version);
@@ -138,7 +130,6 @@ public class VersionConverter {
 
         // Partial semver
         long numberOfDots = version.chars().filter(ch -> ch == '.').count();
-        System.out.println(">>> Partial " + version);
         if (!version.startsWith(EQUAL_TO) && numberOfDots < 2) {
             return convertPartialSemver(version);
         }

--- a/src/main/java/io/mvnpm/version/VersionConverter.java
+++ b/src/main/java/io/mvnpm/version/VersionConverter.java
@@ -26,6 +26,7 @@ import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
+import java.util.stream.Collectors;
 
 import org.jboss.logging.Logger;
 
@@ -47,6 +48,7 @@ public class VersionConverter {
     }
 
     public static String convert(String versionString) {
+        System.out.println("> CONVERT " + versionString);
         try {
             if (null == versionString || versionString.startsWith("git:/") || versionString.startsWith("git+http")) { // We do not support git repos as version. Maybe something we can add later
                 versionString = EMPTY;
@@ -69,8 +71,24 @@ public class VersionConverter {
     private static String convertMultiple(String[] versions) {
         List<String> versionList = new ArrayList<>();
         for (String v : versions) {
-            versionList.add(convertMultiplePart(v.trim()).trim());
+            System.out.println(">> MULTIPLE " + v);
+            String result = convertMultiplePart(v.trim()).trim();
+            System.out.println("<< result " + result);
+            versionList.add(result);
         }
+
+        // If multiple, it has to be in brackets
+        if (versionList.size() > 1) {
+            versionList = versionList.stream()
+                    .map(element -> {
+                        if (!element.contains("[") && !element.contains("(") && !element.contains("]")
+                                && !element.contains(")")) {
+                            return "[" + element + "]";
+                        }
+                        return element;
+                    }).collect(Collectors.toList());
+        }
+
         return String.join(COMMA, versionList);
     }
 
@@ -81,39 +99,46 @@ public class VersionConverter {
      * @return
      */
     private static String convertMultiplePart(String version) {
+
         // Hyphen range
         if (version.contains(SPACE + HYPHEN + SPACE)) {
+            System.out.println(">>> Hyphen " + version);
             return convertHyphen(version);
         }
 
         // Tilde range
         if (version.startsWith(TILDE)) {
+            System.out.println(">>> Tilde " + version);
             version = cleanTildeVersion(version);
             return convertTilde(version);
         }
 
         // Caret range
         if (version.startsWith(CARET)) {
+            System.out.println(">>> Caret " + version);
             version = cleanCaretVersion(version);
             return convertCaret(version);
         }
 
+        // Operator range
+        if (version.startsWith(LESS_THAN) || version.startsWith(GREATER_THAN)) {
+            System.out.println(">>> Operator " + version);
+            version = cleanOperatorVersion(version);
+            return convertOperator(version);
+        }
+
         // X range
         if (!version.contains(SPACE)) {
+            System.out.println(">>> Xrange " + version);
             String numberPart = version.split(HYPHEN)[0];
             if (numberPart.contains(STAR) || numberPart.contains(EX) || numberPart.contains(EX.toUpperCase())) {
                 return convertX(version);
             }
         }
 
-        // Operator range
-        if (version.startsWith(LESS_THAN) || version.startsWith(GREATER_THAN)) {
-            version = cleanOperatorVersion(version);
-            return convertOperator(version);
-        }
-
         // Partial semver
         long numberOfDots = version.chars().filter(ch -> ch == '.').count();
+        System.out.println(">>> Partial " + version);
         if (!version.startsWith(EQUAL_TO) && numberOfDots < 2) {
             return convertPartialSemver(version);
         }

--- a/src/test/java/io/mvnpm/VersionConverterTest.java
+++ b/src/test/java/io/mvnpm/VersionConverterTest.java
@@ -275,6 +275,12 @@ public class VersionConverterTest {
         Assertions.assertEquals("[1,)", mavenVersion);
     }
 
+    @Test
+    public void testGreaterThanOrEqualToVersionXMinor() {
+        String mavenVersion = VersionConverter.convert(">=14.x");
+        Assertions.assertEquals("[14,)", mavenVersion);
+    }
+
     // Greater than or equalTo AND Less than or equalTo
     @Test
     public void testGreaterThanOrEqualToAndLessThanOrEqualToVersionWithZeroMajor() {
@@ -491,13 +497,19 @@ public class VersionConverterTest {
     @Test
     public void testOr() {
         String mavenVersion = VersionConverter.convert("1.2.7 || >=1.2.9 <2.0.0");
-        Assertions.assertEquals("1.2.7,[1.2.9,2.0.0)", mavenVersion);
+        Assertions.assertEquals("[1.2.7],[1.2.9,2.0.0)", mavenVersion);
+    }
+
+    @Test
+    public void testGreaterThanOrEqualToORVersion() {
+        String mavenVersion = VersionConverter.convert(">= 16.8 || 18.0.0");
+        Assertions.assertEquals("[16.8,),[18.0.0]", mavenVersion);
     }
 
     @Test
     public void testOrOr() {
         String mavenVersion = VersionConverter.convert("1.2.7 || >=1.2.9 <2.0.0 || >3.3.2 <=4.0.0");
-        Assertions.assertEquals("1.2.7,[1.2.9,2.0.0),(3.3.2,4.0.0]", mavenVersion);
+        Assertions.assertEquals("[1.2.7],[1.2.9,2.0.0),(3.3.2,4.0.0]", mavenVersion);
     }
 
     // Partial version in range

